contents/tutorials/sticky-feature-flags.md

```diff
@@ -0,0 +1,208 @@
+---
+title: How to create sticky feature flags
+date: 2025-06-16
+author:
+  - dylan-martin
+showTitle: true
+sidebar: Docs
+tags:
+  - feature flags
+  - experiments
+---
+
+Sticky feature flags ensure users maintain their original variant assignment even after rollout conditions change. This is crucial for experiments where you need consistent user experiences throughout the test period, regardless of changes to targeting rules or rollout percentages.
+
+This tutorial shows how to set up sticky feature flags in PostHog by combining person properties with feature flag conditions to guarantee consistent variant assignments.
+
+## The problem with regular feature flags
+
+Regular feature flags can reassign users to different variants when you modify rollout conditions. For example:
+
+- You start an A/B experiment with 50% of users seeing variant A
+- Later, you decrease it to 25% rollout
+- Some users who previously saw variant A might now see variant B
+
+This inconsistency can skew your experiment results and create a poor user experience.
+
+## How sticky flags work
+
+Sticky flags solve this by:
+
+1. **Setting a person property** when a user first evaluates the flag
+2. **Adding override conditions** to the flag that check for existing assignments
+3. **Guaranteeing consistency** by prioritizing the stored assignment over new rollout rules
+
+## Setting up sticky feature flags
+
+### Step 1: Create your feature flag
+
+First, create your feature flag normally in PostHog:
+
+1. Go to [Feature Flags](https://app.posthog.com/feature_flags) and click "New feature flag"
+2. Set your flag key (e.g., `sticky-experiment`)
+3. Configure your initial rollout conditions
+4. Save the flag
+
+### Step 2: Capture person properties on evaluation
+
+When evaluating the flag in your code, capture the result as a person property. Here's an example using the JavaScript SDK:
+
+**Important:** Use `$set_once` instead of `$set` to ensure the property is only set on the first evaluation and never overwritten.
```

Comment:
```suggestion
import { CalloutBox } from 'components/Docs/CalloutBox'

<CalloutBox icon="IconInfo" title="Important" type="fyi">
  
Use `$set_once` instead of `$set` to ensure the property is only set on the first evaluation and never overwritten.

</CalloutBox>
```